doc/whatsnew/fragments/10100.false_positive

```diff
@@ -0,0 +1,4 @@
+Fix a false positive for `use-implicit-booleaness-not-len`. No lint should be emitted for
+generators (`len` is not defined for generators).
+
+Closes #10100
```

Comment:
```suggestion
Refs #10100
```
I feel like to close #10100  a message should be raised about the type error (add the ``if isinstance(len_arg, (nodes.GeneratorExp)):`` condition and raise the proper message in this case), because nothing is raised at the moment. But I don't know what to use in the existing list of message (probably should create a new message like "invalid-len-call" ?). What do everyone think ?

Reply:
We could certainly perform the check within the current `ImplicitBooleanessChecker`, but I believe it would make more sense to create a dedicated checker for invalid uses of `len`. The reason for this is that raising an `invalid-len-call` message in a class designed for `ImplicitBooleaness` might lead to confusion or unnecessary coupling of unrelated concerns. By creating a separate checker, we can more clearly handle cases in which `len` is used improperly while keeping the scope of the existing checker focused on its intended task.

Reply:
Should I implement the new checker, maybe adding it to `pylint/checkers` with the name `invalid_calls`?

Reply:
Thank you for proposing. It's an error level message so it ~~could~~ should (?) be in a builtin checker really (BasicErrorChecker maybe ?). What do @jacobtylerwalls or @DanielNoord think ?